### PR TITLE
add a db_control option to intersect_featuremap_with_signature

### PR DESCRIPTION
**Pull request by GatKrieger**
needed for updating mrd_featuremap.wdl

### CODE REVIEW

**Comment by GatKrieger** [comment]
I need help with changing the --is_matched flag in other functions: read_signature, featuremap_to_dataframe

**Review by kenissur — APPROVED** [review]

File: ugvc/mrd/mrd_utils.py [hunk 1]
```diff
@@ -92,6 +92,16 @@ def _safe_tabix_index(vcf_file: str):
     return vcf_file
 
 
+def _validate_info(x):
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py:95]: First check if the input is boolean and if so return it, then checks whether it's a string before doing string operations, and please add a docstring and explain what this is for.

File: ugvc/mrd/mrd_utils.py [hunk 2]
```diff
@@ -209,6 +219,7 @@ def read_signature(  # pylint: disable=too-many-branches
             logger.debug(f"Reading x_columns: {x_columns}")
 
             for j, rec in enumerate(variant_file):
+                # print(rec)
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py]: remove

File: ugvc/mrd/mrd_utils.py [hunk 3]
```diff
@@ -235,7 +246,8 @@ def read_signature(  # pylint: disable=too-many-branches
                             else np.nan,
                             rec.info["SOR"] if "SOR" in rec.info else np.nan,
                         ]
-                        + [c in rec.info for c in genomic_region_annotations]
+                        # [c in rec.info for c in genomic_region_annotations]
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py]: remove commented line

File: ugvc/mrd/mrd_utils.py [hunk 4]
```diff
@@ -301,7 +313,7 @@ def read_signature(  # pylint: disable=too-many-branches
                 for variant_file in f_bw:
                     variant_file.close()
         df_sig = pd.concat(df_list).astype({"coverage": int})
-
+    df_sig["hmer"] = np.nan  # it read TRUE/FALSE in some cases, so we reset it to nan
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py]: 1. Please explain
2. I don't like this patch, if the issue is that sometime you get string values then find a way to force-convert to float (as in, convert to float if possible and it not set to NaN), pretty sure pandas can do that

File: ugvc/mrd/mrd_utils.py [hunk 5]
```diff
@@ -808,22 +787,28 @@ def prepare_data_from_mrd_pipeline(
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py:826]: Please make sure at least one of db_control_signatures_vcf_files, control_signatures_vcf_files, matched_control_signatures_vcf_files is not None
Recommend adding a test that gives just one of these to make sure it doesn't fail

File: ugvc/mrd/mrd_utils.py [hunk 6]
```diff
@@ -93,13 +93,24 @@ def _safe_tabix_index(vcf_file: str):
...
+    """
+    if isinstance(x, bool):
+        return x
+    if isinstance(x, str):
+        if x.lower() == "true":
+            return True
+        if x.lower() == "false":
+            return False
+        if x.isnumeric():
+            return True
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py]: if x==0 then I would expect False, so maybe "return bool(float(x))"?

**Pull request merged**